src/scheduler/server_info.c

```diff
@@ -280,7 +280,7 @@ query_server(status *pol, int pbs_sd)
 	if ((sinfo->queues = query_queues(policy, pbs_sd, sinfo)) == NULL) {
 		pbs_statfree(server);
 		sinfo->fairshare = NULL;
-		free_server(sinfo, 0);
+		free_server(sinfo, 1);
```

Comment:
Not sure if there was a particular reason why this was called with a 0, but if we are changing this, then shouldn't we also change the other free_server() calls in this function?

Reply:
the subsequent calls are already using 1, instead of 0.
The calls before this need not change as when called with 1 - free_server() would free queues, nodes and resource_resv_array. Any of these have not been allocated until this point.

Reply:
I'll suggest to just remove the option to free only certain things in free_server()
It does not make sense to free only partial things. I understand that free will potentially be called on unallocated things but if they are initialized properly (with NULL) it should not be a problem. This will also reduce the possibility of future problems with free_server()

Reply:
I thought about the same - but chose to be lazy and do it if this is suggested as a review comment :)

Reply:
Well now you have it 😄